core/src/main/java/com/graphhopper/routing/profiles/BooleanEncodedValue.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.routing.profiles;
+
+import com.graphhopper.storage.IntsRef;
+
+/**
+ * This class provides easy access to just one bit.
+ */
+public final class BooleanEncodedValue extends IntEncodedValue {
+
+    /**
+     * The default value is false.
+     */
+    public BooleanEncodedValue(String name) {
+        this(name, false);
+    }
+
+    public BooleanEncodedValue(String name, boolean store2DirectedValues) {
+        super(name, 1, 0, store2DirectedValues);
+    }
+
+    public final void setBool(boolean reverse, IntsRef ref, boolean value) {
+        if (store2DirectedValues && reverse) {
+            int flags = ref.ints[bwdDataIndex + ref.offset];
+            flags &= ~bwdMask;
+            // set value
+            if (value)
+                flags = flags | (1 << bwdShift);
+            ref.ints[bwdDataIndex + ref.offset] = flags;
+
+        } else {
+            int flags = ref.ints[fwdDataIndex + ref.offset];
+            flags &= ~fwdMask;
+            if (value)
+                flags = flags | (1 << fwdShift);
+            ref.ints[fwdDataIndex + ref.offset] = flags;
+        }
```

Comment:
The if and else branch look pretty similar, maybe we can combine them in one function?

Reply:
This is a very perf critical section and avoiding code duplication seems only possible with some additional variables

Reply:
It's only a couple of lines, so it's not a big issue anyway. I thought we might be able to rewrite the code to be something like:

```
        if (store2DirectedValues && reverse) {
            _setBool(bwdDataIndex, bwdMask, value, bwdShift)
        } else {
            _setBool(fwdDataIndex, fwdMask, value, fwdShift)
        }
```

Reply:
I wouldn't risc this for now